modules/localstack/examples_test.go

```diff
@@ -113,3 +116,83 @@ func ExampleRunContainer_legacyMode() {
 	// Output:
 	// version=localstack/localstack:0.10.0. Testcontainers for Go does not support running LocalStack in legacy mode. Please use a version >= 0.11.0
 }
+
+func ExampleRunContainer_UsingLambdas() {
+	ctx := context.Background()
+
+	flagsFn := func() string {
+		labels := testcontainers.GenericLabels()
+
+		flags := ""
+		for k, v := range labels {
+			flags = fmt.Sprintf("%s -l %s=%s", flags, k, v)
+		}
+
+		return flags
+	}
+
+	lambdaName := "localstack-lambda-url-example"
+
+	container, err := localstack.RunContainer(ctx,
+		testcontainers.WithImage("localstack/localstack:2.3.0"),
+		testcontainers.CustomizeRequest(testcontainers.GenericContainerRequest{
+			ContainerRequest: testcontainers.ContainerRequest{
+				Env: map[string]string{
+					"SERVICES":                 "lambda",
+					"LAMBDA_EXECUTOR":          "docker",
+					"PROVIDER_OVERRIDE_LAMBDA": "asf",
+					"LAMBDA_DOCKER_FLAGS":      flagsFn(),
+				},
+				Files: []testcontainers.ContainerFile{
+					{
+						HostFilePath:      filepath.Join("testdata", "function.zip"),
+						ContainerFilePath: "/tmp/function.zip",
+					},
+				},
+			},
+		}),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer container.Terminate(ctx)
+
+	// the three commands below are doing the following:
+	// 1. create a lambda function
+	// 2. wait for the lambda function to be active
+	// 3. invoke the lambda function with a payload, writing the result to the output.txt file
+	lambdaCommands := [][]string{
+		{
+			"awslocal", "lambda",
+			"create-function", "--function-name", lambdaName,
+			"--runtime", "nodejs18.x",
+			"--zip-file",
+			"fileb:///tmp/function.zip",
+			"--handler", "index.handler",
+			"--role", "arn:aws:iam::000000000000:role/lambda-role",
+		},
+		{"awslocal", "lambda", "wait", "function-active-v2", "--function-name", lambdaName},
+		{"awslocal", "lambda", "invoke", "--function-name", lambdaName, "--payload", `{"body": "{\"num1\": \"10\", \"num2\": \"10\"}" }`, "output.txt"},
+	}
+	for _, cmd := range lambdaCommands {
+		_, _, err = container.Exec(ctx, cmd)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// the output.txt file lives in the WORKDIR of the localstack container
+	_, reader, err := container.Exec(ctx, []string{"cat", "output.txt"}, exec.Multiplexed())
+	if err != nil {
+		panic(err)
+	}
+
+	content, err := io.ReadAll(reader)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(content))
```

Comment:
What about asserting the text returned by the function?

Reply:
In testable examples, there are no assertions but an internal comparison of the output of stdout against [the](https://github.com/testcontainers/testcontainers-go/pull/1790/files#diff-2a7f31c70cbb0499942309ff81dbc15898810fb8734f0278eaa52f4da525b16dR196-R197)